include/tarxx.h

```diff
@@ -424,6 +491,54 @@ namespace tarxx {
             return str;
         }
 
+        void write_name_and_prefix(block_t& block, const std::string& name)
+        {
+            const auto write_name_unix_v7_format = [&]() {
+                write_into_block(block, name, UNIX_V7_USTAR_HEADER_POS_NAME, UNIX_V7_USTAR_HEADER_LEN_NAME);
+            };
+
+            if (name.size() <= UNIX_V7_USTAR_HEADER_LEN_NAME || type_ == tar_type::unix_v7) {
+                write_name_unix_v7_format();
+            } else {
+                const auto last_slash_index = name.rfind('/');
```

Comment:
separating at the last slash has one and a half problems:
1. on other platforms the directory separator might be different, e.g. Windows uses a backslash (`\`) - this probably needs to go into the OS abstraction layer.
2. tar doesn't mandate separating at the _last_ slash, but at _some_ slash (or more precisely: at _a_ slash only). there could be cases where would have to separate the whole path+filename because it doesn't fit in the v7 field, but using the last slash just won't do the trick. like so `insanely_long_path_name_in_front_.../and_yet_another_very_long_path_component_.../my.file` ...

Reply:
take a look at the last commit, should be fixed now